### PR TITLE
update the readme

### PR DESCRIPTION
**Pull request by devoncarew**
- update the readme - add additional docs as well as a usage example
- fix https://github.com/dart-lang/web/issues/107

Please feel free to provide feedback in the form of direct edits here -

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,34 @@
...
-This package will *eventually* replace [`dart:html`](https://api.dart.dev/stable/dart-html/dart-html-library.html) and
-similar libraries in the Dart SDK to fully support WebAssembly.
+## What's this?
+
+This package exposes the browser's DOM APIs. It's generated from the Web IDL
+definitions and uses recent Dart language features for zero-overhead bindings.
+
+This package is intended to replace
+[`dart:html`](https://api.dart.dev/stable/dart-html/dart-html-library.html) and
+similar Dart SDK libraries. The package will support DOM access from Dart code
```
Comment (kevmoo) [inline README.md:14]:
```suggestion
similar Dart SDK libraries. It will support DOM access from Dart code
```

File: README.md [hunk 2]
```diff
@@ -4,9 +4,34 @@
...
+compiled to either JavaScript or WebAssembly.
+
+## Status
+
+The APIs in this package are still fairly provisional and could change rapidly.
+We'll continue to use semver as we evolve the APIs.
```
Comment (kevmoo) [inline README.md:20]: Maybe `Semantic Versioning` with a link to the spec?

**Review by kevmoo — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -4,9 +4,34 @@
 
 Lightweight browser API bindings built around JS static interop.
 
-This package will *eventually* replace [`dart:html`](https://api.dart.dev/stable/dart-html/dart-html-library.html) and
-similar libraries in the Dart SDK to fully support WebAssembly.
+## What's this?
+
+This package exposes the browser's DOM APIs. It's generated from the Web IDL
```
Comment (srujzs) [inline README.md:9]:
```suggestion
This package exposes browser APIs. It's generated from the Web IDL
```

We expose more than just the DOM APIs.

**Review by srujzs — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (kevmoo) [reply README.md:9]: true true!

File: README.md (hunk 2 above)
Comment (devoncarew) [reply README.md]: 👍  - done

**Comment by kevmoo** [comment]
ship it1

**Pull request merged**